var/spack/repos/builtin/packages/cuda/package.py

```diff
@@ -51,6 +51,10 @@ class Cuda(Package):
             url="http://developer.download.nvidia.com/compute/cuda/7.5/Prod/local_installers/cuda_7.5.18_linux.run")
     version('6.5.14', '90b1b8f77313600cc294d9271741f4da', expand=False,
             url="http://developer.download.nvidia.com/compute/cuda/6_5/rel/installers/cuda_6.5.14_linux_64.run")
+    # use up to gcc@4 when using cuda@8 
+    # use up to gcc@6 when using cuda@9
+    conflicts('%gcc@7:', when='@9:', msg='use cuda less than version 7')
```

Comment:
have you seen https://github.com/spack/spack/blob/develop/lib/spack/spack/build_systems/cuda.py#L65-L78 ?